x/concentrated-liquidity/types/keys.go

```diff
@@ -113,10 +116,30 @@ func keyTickPrefixByPoolIdPrealloc(poolId uint64, preAllocBytes int) []byte {
 	return key
 }
 
-// KeyFullPosition uses pool Id, owner, lower tick, upper tick, joinTime and freezeDuration for keys
-func KeyFullPosition(poolId uint64, addr sdk.AccAddress, lowerTick, upperTick int64, joinTime time.Time, freezeDuration time.Duration) []byte {
+// KeyFullPosition uses pool Id, owner, lower tick, upper tick, joinTime, freezeDuration, and positionId for keys
+func KeyFullPosition(poolId uint64, addr sdk.AccAddress, lowerTick, upperTick int64, joinTime time.Time, freezeDuration time.Duration, positionId uint64) []byte {
 	joinTimeKey := osmoutils.FormatTimeString(joinTime)
-	return []byte(fmt.Sprintf("%s%s%x%s%d%s%d%s%d%s%s%s%d", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator, poolId, KeySeparator, lowerTick, KeySeparator, upperTick, KeySeparator, joinTimeKey, KeySeparator, uint64(freezeDuration)))
+
+	var builder strings.Builder
```

Comment:
Consider using `bytes.Buffer` instead of `strings.Builder`.

When you need to serialize data consisting of strings, byte slices, integers, and time in Go, it is more efficient to use bytes.Buffer. The reason is that bytes.Buffer can handle both strings and byte slices directly, whereas strings.Builder is designed to work only with strings.

Using bytes.Buffer allows you to avoid unnecessary conversions between strings and byte slices, which can lead to better performance and reduced memory overhead.

Reply:
Also, I know my issue https://github.com/osmosis-labs/osmosis/issues/4626 suggested `strings.Builder` but I thought about this only upon seeing this change so apologies for re-requesting a change I earlier suggested